notes-app-server/src/services/notes.service.ts

```diff
@@ -0,0 +1,25 @@
+import { DatabasePort } from "../persist/database.port";
+import { Note, NoteWithId } from "../validation/note.schema";
+
+export class NotesService {
+  private database: DatabasePort;
+  constructor(database: DatabasePort) {
```

Comment:
Notice here the `NotesService` does not need to know how the `DatabasePort` is implemented. It accepts `DatabasePort` - any implementation will do as long as it's compliant with the port.

The responsibility of the notes service is not to "store it in the database using Prisma ORM".
The responsibility is to perform "CRUD (Create, Read, Update, Delete) operations for note using database).

Here the functionality is very thin, so there is not much of a difference, but it could also encompass more complex logic such as transforming note content (e.g., markdown to HTML conversion), managing note relationships (e.g., tags, categories), and integrating with other services or components.